### DISCUSSION TITLE
Recommendations for the Largest SPI Supported Screen ?

### DISCUSSION THREAD

**Discussion by bionicbone** [Bodmer/TFT_eSPI#1810] [Compatible displays and setup files]
As the title suggests, has anyone got any recommendations for the largest screen with SPI support?

I am not being lazy honest, I've spent hours searching AliExpress and BuyDisplays but most that I see are either Parallel or use the RA8875 chip that is not supported.

Parallel is not an option as my ESP32 is already running low on pins :(

7" would have been ideal, 5" a make do, smaller and I'll need to think again how I do this.

I except refresh rates will be low over SPI on a large screen, however for my application once the screen is completed then only needles or small data updates will be required to change.

I'll need to share the SPI interface with two other 3" displays, thus Cable Select will need to work.

**Comment by Bodmer** [comment]
The SSD1963 is the only option I am aware of and have tested, but it is 8 bit parallel only, not SPI.

The main reason for not supporting other large screens is cost. It rapidly gets to the point where it is more cost effective to use a cheap Android tablet or phone as the display.

Above 4" I use the ESP32 as the sensor on a WiFi network and a RPi Node-RED server, so I can see the data on any networked  device with a web browser. This means a cheap Android tablet or phone can also be used as the screen. This is a more complicated and complex system but a good use of old or cheap tablets. Being able to see the sensor data on my phone or PC screen at any time is also handy.

A second ESP32 could also be used as the display device.

**Reply by bionicbone** [reply]
Good point, I had not thought of the WiFi and Android idea.

"A second ESP32 could also be used as the display device"   this is my thoughts now though. 

Thanks for your input.

**Reply by Bodmer** [reply]
The Pico (RP2040) processor is very good for parallel displays. The RP2040 PIO is used by the library  so screen clearing is much quicker than an ESP32 can manage.

**Reply by bionicbone** [reply]
Thanks for that, another bit of great info, I'll order one so it's here for when the screen arrives.

**Comment by XFer012** [comment]
I am in a similar situation. I need the ESP32 to drive a large (5" or better 7") display; performances are not important (mostly static output, mainly text).
I see SSD1963-based displays are supported: how to wire them to the ESP32? 
Besides: any other caveats/suggestions/issues? 

I see I could "wire" a RP2040 to the display and control it from the ESP32, but again: any known good large 5"-7" display which plays well with RP2040?

Thanks

**Reply by Bodmer** [reply]
This is the one I have:
https://www.buydisplay.com/7-tft-screen-touch-lcd-display-module-w-ssd1963-controller-board-mcu

8080 8 bit parallel interface, 5V (regulator onboard to 3.3V), capacitance touch (ar better than resistive base screens) which uses an I2C bus. 

I used an 8 bit interface to the ESP32.  GPIO connections here.
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup50_SSD1963_Parallel.h

I created a SPI touch library for it here:
https://github.com/Bodmer/Touch_FT5x06

But there are other libraries for the capacitance touch controller.

**Reply by XFer012** [reply]
Thanks! I don't need touch right now, but good to know

**Reply by pipikupi** [reply]
Hi I'm using this one :
https://www.ebay.co.uk/itm/155736909997?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=XTMPZF_FQCG&sssrc=4429486&ssuid=XpkRFA37Qde&var=&widget_ver=artemis&media=COPY
It's SSD1963 works very well at 8 bit interface.
![IMG_20231007_143302.jpg](https://github.com/Bodmer/TFT_eSPI/assets/76049929/9479e0bd-d4d2-4f70-b3ae-a548e529a2d9)

But it is more static work.
Project based on Bodmer's Openweather map example.

**Reply by Bodmer** [reply]
The ebay advert seens a bit vague about the controller, an RA8875 is also mentioned along with the SSD1963, so it may be a bit of a lottery which one you get?  The RA8875 is not supported by TFT_eSPI.

**Comment by phanthaitrung** [comment]
As I see my ESP32-S3 supports 16 bit parallel transfer, @Bodmer do you think adding option for 16 bit instead of current 8 bit can increase transfer speed? Is it possible?

Off topic I got the 8inch screen 4-wire SPI with LT7683 driver chip from Buydisplay, it works with their internal Arduino lib, but I think SPI is so slow with large 8 inch 800x600 and my option is thinking of ordering another screen with parallel transfer instead. SSD1963 is prefer instead of LT7683 because TFT_eSPI currently support SSD1963 but not LT7683 yet.

**Reply by Bodmer** [reply]
16 bit parallel is only supported with RP2040 processors.

There are no plans to add 16 bit parallel support to other processors at the moment.
